### ISSUE TITLE
Compilation warnings in RecoLocalCalo/EcalRecProducers

### ISSUE THREAD

**Issue by Dr15Jones** [cms-sw/cmssw#14397]
Using gcc 5.3.0 we have the compilation warning:

```
  /build/cmsbuild/jenkins-workarea/workspace/build-any-ib/w/slc6_amd64_gcc530/cms/cmssw/CMSSW_8_1_X_2016-05-01-2300/src/RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h:69:36: warning: comparison of constant '-1' with boolean expression is always true [-Wbool-compare]
        if ( dataFrame.isSaturated() != -1 ) 
                                    ^
```

**Comment by Dr15Jones** [comment]
assign reconstruction

**Comment by Dr15Jones** [comment]
This was seen here:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc530/CMSSW_8_1_X_2016-05-05-2300/RecoLocalCalo/EcalRecProducers

**Comment by slava77** [comment]
+1

fixed in #17107
